src/Simulation/Common/QubitManager.cs

```diff
@@ -43,6 +43,7 @@ public class QubitManager : IQubitManager
         private long AllocatedForBorrowing; // All qubits allocated only for borrowing, will be marked with this number or higher.
         private long free; // Points to the first free (unallocated) qubit.
         private long freeTail; // Points to the last free (unallocated) qubit. Only valid iff (!EncourageReuse).
+        private static HashSet<Qubit> EMPTY_SET = new HashSet<Qubit>();
```

Comment:
Suggest making this readonly and, if possible, create immutable hashset.

Reply:
Removed it. The optimization would be useful when borrowing is disabled to avoid allocating an empty set each time, however, we already check not to call QubitsInUse if borrowing is disabled, and I assume the null frame case isn't that common.

I'd rather not make the returned hashset from QubitsInUse immutable, because we create it afresh on each invocation and at this point, if the caller would like to modify it, they should be able to.